clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -542,8 +544,16 @@ static Map<String, Object> appendSerializerToConfig(Map<String, Object> configs,
         Map<String, Object> newConfigs = new HashMap<>(configs);
         if (keySerializer != null)
             newConfigs.put(KEY_SERIALIZER_CLASS_CONFIG, keySerializer.getClass());
+        else {
+            if (newConfigs.get(KEY_SERIALIZER_CLASS_CONFIG) == null)
+                throw new ConfigException(KEY_SERIALIZER_CLASS_CONFIG + " configuration for KafkaProducer must be non-null.");
+        }
         if (valueSerializer != null)
             newConfigs.put(VALUE_SERIALIZER_CLASS_CONFIG, valueSerializer.getClass());
+        else {
+            if (newConfigs.get(VALUE_SERIALIZER_CLASS_CONFIG) == null)
```

Comment:
Can you use `HashMap#containsKey` if checking for the presence of a value or am I misreading something?

Reply:
Thank you for your suggestion, here we want to check the case where the key exists but the value is null.

Reply:
Oh, I see. So this check covers the situations a) the key is missing and b) the key is there and the value is null, while what I suggested would only cover a). Is there a possibility that you add a comment saying this above the line?

Reply:
sure